tests/benchmarks/FCSBenchmarks/CompilerServiceBenchmarks/FileCascadeBenchmarks.fs

```diff
@@ -37,82 +37,148 @@ module private CascadeProjectHelpers =
         $"""
 module Benchmark0
 let returnValue = 5
-let myFunc0 () = 5"""
+let myFunc0 () = returnValue
+type MyType0 = MyType0 of string
+type MyOtherType0 = MyOtherType0 of int"""
+
+    let baselineFsi = 
+        """
+module Benchmark0
+
+val returnValue: int
+
+val myFunc0: unit -> int
+
+type MyType0 = | MyType0 of string
+
+type MyOtherType0 = | MyOtherType0 of int"""
 
     let generateSourceCode number =
         $"""
 module Benchmark%i{number}
 open Benchmark%i{number-1}
 let myFunc%i{number} () = myFunc%i{number-1}()
+type MyType{number} = MyType{number} of string
+type MyOtherType{number} = MyOtherType{number} of int
+type MyFunctionType{number} = MyType{number} -> MyOtherType{number}
+
+let processFunc{number} (x) (func:MyFunctionType{number}) = 
+    async {{
+        return func(x)
+    }}
```

Comment:
What do you think about adding some more code to make the files more "realistic"? Like I did here https://github.com/dotnet/fsharp/pull/14108

Reply:
I do not want to have fixed code, just repeated. When benchmarking, this could easily fool us into caching what the compiler does and hoping for highly repetitive code (i.e. same function thousand times over).

If we had a solution for a more realistic chain of calls referencing each other, that could work.